From dad0b8fa43e3348ca0c11e8489c21473264eedb0 Mon Sep 17 00:00:00 2001
From: Patrick Lower <devvortex@gmail.com>
Date: Mon, 4 Jan 2016 11:32:17 -0500
Subject: [PATCH 08/12] MMS: Update apnProfileID for MMS only apn.

Based on system property (ro.telephony.mms_data_profile) set at device level

Change-Id: I6b4ba90dfc5272e3e81330d479b9347bebfc177f
---
 .../internal/telephony/dataconnection/DcTracker.java        | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index 38d3852b1..61e2b34aa 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -571,6 +571,10 @@ public class DcTracker extends Handler {
 
     private final int mTransportType;
 
+    /** MMS Data Profile Device Override */
+    private static final int MMS_DATA_PROFILE = SystemProperties.getInt(
+            "ro.telephony.mms_data_profile", RILConstants.DATA_PROFILE_DEFAULT);
+
     //***** Constructor
     public DcTracker(Phone phone, int transportType) {
         super();
@@ -4029,6 +4033,8 @@ public class DcTracker extends Handler {
             return RILConstants.DATA_PROFILE_DEFAULT; // DEFAULT for now
         } else if (TextUtils.equals(apnType, PhoneConstants.APN_TYPE_DUN)) {
             return RILConstants.DATA_PROFILE_TETHERED;
+        } else if (TextUtils.equals(apnType, PhoneConstants.APN_TYPE_MMS)) {
+            return MMS_DATA_PROFILE;
         } else {
             return RILConstants.DATA_PROFILE_DEFAULT;
         }
-- 
2.17.1

